src/Components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -346,13 +352,18 @@ const ResourcePicker: React.FC<IResourcePickerProps> = ({
 
     useEffect(() => {
         if (selectedOptionProp) {
-            setSelectedOption({
-                key: selectedOptionProp,
-                text: selectedOptionProp,
-                styles: comboBoxOptionStyles
-            });
-        } else {
-            setSelectedOption(null);
+            const existingOption = options?.find((o) =>
+                areValuesEqual(o.text, selectedOptionProp)
+            );
+            if (existingOption) {
+                setSelectedOption(existingOption);
+            } else {
+                setSelectedOption({
+                    key: selectedOptionProp,
+                    text: selectedOptionProp,
+                    styles: comboBoxOptionStyles
+                });
+            }
         }
     }, [selectedOptionProp]);
```

Comment:
do we expect the prop to change? Could we just store teh values on mount and not need the compleixty of maintaining both the value from prop as well as state?

Reply:
yes we expect, it happens when we change the values in dropdown by selecting another option but not save and click to close the modal. In this case we need to revert the selected options back to the original/initial selected values